backend/api/Repositories/RestCommon/BaseRestRespository.cs

```diff
@@ -0,0 +1,212 @@
+                {
+                    Status = ExternalResultStatus.Error,
+                    Message = "Exception during Put",
+                };
+            }
+        }
+
+        public async Task<ExternalResult<string>> DeleteAsync(Uri endpoint, string authenticationToken = null)
+        {
+            using HttpClient client = _httpClientFactory.CreateClient("Pims.Api.Logging");
+            client.DefaultRequestHeaders.Accept.Clear();
+            AddAuthentication(client, authenticationToken);
+            client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue(MediaTypeNames.Application.Json));
+
+            ExternalResult<string> result = new ExternalResult<string>()
+            {
+                Status = ExternalResultStatus.Error,
+                Payload = endpoint.AbsolutePath,
+            };
+
+            try
+            {
+                HttpResponseMessage response = await client.DeleteAsync(endpoint).ConfigureAwait(true);
+
+                _logger.LogTrace("Response: {response}", response);
+
+                string payload = await response.Content.ReadAsStringAsync().ConfigureAwait(true);
+                result.HttpStatusCode = response.StatusCode;
+                switch (response.StatusCode)
+                {
+                    case HttpStatusCode.OK:
+                        _logger.LogTrace("Response payload: {payload}", payload);
+                        result.Status = ExternalResultStatus.Success;
+                        break;
+                    case HttpStatusCode.NoContent:
+                        result.Status = ExternalResultStatus.Success;
+                        result.Message = "No content was returned from the call";
+                        break;
+                    case HttpStatusCode.Forbidden:
+                        result.Status = ExternalResultStatus.Error;
+                        result.Message = "Request was forbidden";
+                        break;
+                    case HttpStatusCode.BadRequest:
+                        result.Status = ExternalResultStatus.Error;
+                        result.Message = payload;
+                        break;
+                    default:
+                        result.Status = ExternalResultStatus.Error;
+                        result.Message = $"Unable to contact endpoint {response.RequestMessage.RequestUri}. Http status {response.StatusCode}";
+                        break;
+                }
+                return result;
+            }
+            catch (Exception e)
+            {
+                result.Status = ExternalResultStatus.Error;
+                result.Message = "Exception during Delete";
+                _logger.LogError("Unexpected exception during delete {e}", e);
+            }
+            return result;
+        }
+
+        private async Task<ExternalResult<T>> ProcessResponse<T>(HttpResponseMessage response)
+        {
+            ExternalResult<T> result = new ExternalResult<T>()
```

Comment:
This looks like it was all done before we discussed using ProblemDetails so fine to leave until we do 4468

Reply:
I dont think is the same thing. The conversation around ProblemDetails was about handling exceptions at the application level and improving them with additional information. Results from external calls should be propagated from the repositories.